app/controllers/ReportingController.scala

```diff
@@ -87,6 +95,59 @@ trait ReportingController extends BaseController {
     }
   }
 
+  private def makeRow(values: Option[String]*) =
+    values.map { s =>
+      val ret = s.getOrElse(" ").replace("\r", " ").replace("\n", " ").replace("\"", "'")
+      "\"" + ret + "\""
+    }.mkString(",")
+
+  def assessorAllocationReport: Action[AnyContent] = Action.async { implicit request =>
+
+    import common.Joda._
+
+    val sortedEventsFut = eventsRepository.findAll.map(_.sortBy(_.date))
+
+    val reportRows = for {
+      allAssessors <- assessorRepository.findAll
+      allAssessorsPersonalInfo <- authProviderClient.findByUserIds(allAssessors.map(_.userId)).map(_.groupBy(_.userId).mapValues(_.head))
```

Comment:
you could reduce the ```groupBy(_.userId).mapValues(_.head)``` bit to ```map(x => x.userId -> x)(breakOut)``` to avoid the extra collection traversals, I think